core/src/main/java/org/apache/spark/memory/TaskMemoryManager.java

```diff
@@ -195,9 +197,9 @@ public long acquireExecutionMemory(long required, MemoryConsumer requestingConsu
       }
 
       consumers.add(requestingConsumer);
-      if (logger.isDebugEnabled()) {
-        logger.debug("Task {} acquired {} for {}", taskAttemptId, Utils.bytesToString(got),
-          requestingConsumer);
+      if (LOGGER.isDebugEnabled()) {
+        LOGGER.debug("Task {} acquired {} for {}", String.valueOf(taskAttemptId),
+          Utils.bytesToString(got), requestingConsumer.toString());
```

Comment:
This place does not need to check whether `requestingConsumer` is null, because it has been checked previously as following:
<img width="501" alt="image" src="https://github.com/apache/spark/assets/15246973/e2284d90-ae20-48ec-92ff-4f91b4a111ae">